### PR TITLE
Fix npm run build:thumbnails

### PR DESCRIPTION
**Pull request by erikdubbelboer**
Building thumbnails fails for `ClusteredLightingExample` and `MiniStatsExample` because they have `ENGINE = 'PERFORMANCE'` which doesn't work with thumbnail generation and fails with

```
Failed to fetch dynamically imported module: http://localhost:12321/iframe/playcanvas.prf.js
```

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by kungfooman** [comment]
Thank you very much for pointing this out!

Now this points to a deeper issue, namely that `npm run build` is supposed to build all three targets:
 - dist/iframe/playcanvas.js
 - dist/iframe/playcanvas.dbg.js
 - dist/iframe/playcanvas.prf.js

Because every example is able to choose the one they want:

```js
export class ClusteredLightingExample {
    static CATEGORY = 'Graphics';
    static ENGINE = 'PERFORMANCE';
    static WEBGPU_ENABLED = true;
    static example = example;
}
```

So what I suggest is rewriting `rollup.config.js` a bit at the end:

```js
// We skip building PlayCanvas ourselves when ENGINE_PATH is given.
// In that case we have a watcher which copies all necessary files.
if (ENGINE_PATH === '') {
    /** @type {buildTarget} */
    const pushTarget = (...args) => {
        targets.push(buildTarget(...args));
    };
    if (NODE_ENV === 'production') {
        // Outputs: dist/iframe/playcanvas.js
        pushTarget('release', 'es5', '../src/index.js', 'dist/iframe');
        // Outputs: dist/iframe/playcanvas.dbg.js
        pushTarget('debug', 'es5', '../src/index.js', 'dist/iframe');
        // Outputs: dist/iframe/playcanvas.prf.js
        pushTarget('profiler', 'es5', '../src/index.js', 'dist/iframe');
    } else if (NODE_ENV === 'development') {
        // Outputs: dist/iframe/playcanvas.dbg.js
        pushTarget('debug', 'es5', '../src/index.js', 'dist/iframe');
    } else if (NODE_ENV === 'profiler') {
        // Outputs: dist/iframe/playcanvas.prf.js
        pushTarget('profiler', 'es5', '../src/index.js', 'dist/iframe');
    } else {
        console.warn("NODE_ENV is neither production, development nor profiler.");
    }
}
```

If we don't care to build all three targets (takes some time), we can also just do:

```
ENGINE_PATH=../src/index.js npm run build
```

(nothing is built, only the ESM files are copied into `examples/dist` to reduce time to create thumbnails)

In any case, we need all three targets for production (for `npm run build` + `npm run serve` aswell). Would you mind to edit `rollup.config.js`?

**Comment by erikdubbelboer** [comment]
@kungfooman that also works, I have applied your changes instead 👍

File: examples/rollup.config.js [hunk 1]
```diff
@@ -218,23 +218,25 @@ const targets = [
 // We skip building PlayCanvas ourselves when ENGINE_PATH is given.
 // In that case we have a watcher which copies all necessary files.
 if (ENGINE_PATH === '') {
-    /** @type {RollupOptions|undefined} */
-    let target;
+    const pushTarget = (...args) => {
+        targets.push(buildTarget(...args));
+    };
```
Comment (kungfooman) [inline examples/rollup.config.js:223]:
```suggestion
    /** @type {buildTarget} */
    const pushTarget = (...args) => {
        targets.push(buildTarget(...args));
    };
```

(lets keep the type for full typing / IntelliSense)

**Comment by willeastcott** [comment]
@kungfooman I'm trying to test this PR on my Win10 laptop but `npm run build:thumbnails` doesn't work there:

image

Looks like this is the line to blame:

```js
    const server = spawn('serve', ['dist', '-l', port, '--no-request-logging', '--config', '../serve.json']);
```

This seems to assume `serve` is globally installed maybe? Anyway, the `serve` command isn't found and the process errors out.

**Comment by kungfooman** [comment]
Hey @willeastcott, we got a solution for that here: https://github.com/playcanvas/engine/issues/5888#issuecomment-1848424290

(issue is still open and requires a PR)

**Comment by Maksims** [comment]
Same issue for me. And it used to work, but recently stopped.
Building on Windows also.

**Pull request merged**